hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -196,7 +196,9 @@ protected Dialect() {
 
 		registerColumnType( Types.DATE, "date" );
 		registerColumnType( Types.TIME, "time" );
+		registerColumnType( Types.TIME_WITH_TIMEZONE, "time with time zone" );
```

Comment:
This is PostgreSQL-specific syntax and will break on other database systems. 

I think we should have two overridable methods:

- `protected String getTimeWithTimeZone() { return "time" };`
- `protected String getTimeWithTimeZone() { return "timestamp" };`

Then, each Dialect can override these methods and map the appropriate database-specific types.